HACKING-cli-image.md

```diff
@@ -0,0 +1,98 @@
+Strimzi Development CLI image
+=============================
+
+Strimzi Development CLI image is a CentOS 7 based image that contains Java 8 and all the build dependencies and tools needed to build and deploy Strimzi.
+
+In order to build Strimzi Kafka Operator project you need a running Docker daemon.
+
+If you also want to try them out, deploy the Strimzi Cluster Operator, and run a test Kafka cluster on Kubernetes, you'll need access to a Kubernetes API server.
+
+There are several locally running options for Kubernetes as explained in [Strimzi Quickstarts](https://strimzi.io/quickstarts/).
+
+Look for instructions on installing Docker, Kubernetes in those documents or on web.
+
+For using Strimzi Development CLI image you only need a working Docker daemon, and `docker` client.
+
+
+Starting the session
+--------------------
+
+First, make sure that you have the latest version of the image (best to do this every time you start a new session):
+
+    docker pull quay.io/mstruk/strimzi-dev-cli
+
+Second, you may have to prepare some configurations for accessing Docker daemon, Kubernetes API server, Maven repository ... from within your Docker container session.
+
+For example, if you're using Kind you'll want to export its configuration specifically for use from within Docker containers:
+
+    kind get kubeconfig --internal > ~/.kube/internal-kubeconfig
```

Comment:
To be honest I didn't know about Kind but isn't it possible having the same pipeline using Kubernetes with Minikube? I think it's the most used one from people starting with Kubernetes, but of course I could be wrong.

Reply:
When you're on MacOS you're using Docker Desktop which runs a virtual machine with Linux and Docker daemon. Then if you install Minikube or Minishift, another virtual machine is started with another docker daemon running in it. Having both of these running consumes extra CPU and extra memory. Also, in my experience things are quite fragile when you move around with your laptop and you local network address keeps changing. Even in the middle of normal work the Kubernetes API server often simply becomes unresponsive after some time.

Using Kubernetes Kind I see no such issues. The startup time of Kubernetes is a lot faster making it trivial to discard the cluster and creating it from scratch. Also you can run Docker Registry separately from Kind, and restarting the Kind doesn't scrap the content of the registry so all dependency image layers don't have to be downloaded again.

It's just nicer to work with locally. And kudos to @scholzj for telling me about it :)

Reply:
So it seems Kind provides a better experience for MacOS users. I should check how the experience is for Linux ones :-)

Reply:
Absolutely :)